packages/app/src/AutocompleteInput.tsx

```diff
@@ -213,6 +274,9 @@ export default function AutocompleteInput({
                 suggestedProperties[selectedAutocompleteIndex].value,
               );
             } else {
+              if (queryHistoryType) {
+                setQueryHistory(value);
+              }
               onSubmit?.();
             }
           }
```

Comment:
Just a heads up. In a separate PR, I will add icons for these as part of the larger story for adding icons next to items in the dropdown.

![image](https://github.com/user-attachments/assets/cd91e764-95af-4d03-aa55-d3b3a62a534c)